### PR TITLE
DELAY_STARTUP only when booting

### PR DESCRIPTION
**Pull request by DL6ER**
**By submitting this pull request, I confirm the following:**

- [X] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md).
- [X] I have checked that [another pull request](https://github.com/pi-hole/FTL/pulls) for this purpose does not exist.
- [X] I have considered, and confirmed that this submission will be valuable to others.
- [X] I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
- [X] I give this submission freely, and claim no ownership to its content.

**How familiar are you with the codebase?:** 

## 10

---

Apply `DELAY_STARTUP` only when the system is still booting. The amount of time we consider a system still booting is a compile-time constant. It defaults to 60 seconds.

### CODE REVIEW

File: src/FTL.h [hunk 1]
```diff
@@ -110,6 +110,11 @@
 // Important: This number has to be smaller than 256 for this mechanism to work
 #define NUM_RECHECKS 3
 
+// DELAY_STARTUP should only delay the startup of the resolver during a starting up system
+// This setting control how long after boot we consider a system to be in starting-up mode
+// Default: 600 [seconds]
```
Comment (yubiuser) [inline src/FTL.h:115]:
```suggestion
// Default: 60 [seconds]
```

File: src/daemon.c [hunk 2]
```diff
@@ -173,9 +176,23 @@ void delay_startup(void)
...
+	{
+		logg("Unable to obtain sysinfo: %s (%i)", strerror(errno), errno);
+	}
+
 	// Sleep if requested by DELAY_STARTUP
-	logg("Sleeping for %d seconds as requested by configuration ...",
-	     config.delay_startup);
+	logg("Sleeping for %d seconds as requested by configuration ...", config.delay_startup);
 	sleep(config.delay_startup);
 	logg("Done sleeping, continuing startup of resolver...\n");
```
Comment (yubiuser) [inline src/daemon.c:197]:
```
[2022-05-19 20:19:42.293 1535M] Sleeping for 5 seconds as requested by configuration ...
[2022-05-19 20:19:47.293 1535M] Done sleeping, continuing startup of resolver...

[2022-05-19 20:19:47.438 1535M] Database version is 12
```


```suggestion
	logg("Done sleeping, continuing startup of resolver...");
```

**Review by yubiuser — CHANGES_REQUESTED** [review]
Please file a documentation PR to adjust
https://docs.pi-hole.net/ftldns/configfile/#delay_startup

File: src/daemon.c (hunk 2 above)
Comment (DL6ER) [reply src/daemon.c:197]: I believe this was intentional to get some more attention but this isn't really needed, probably.

**Review by yubiuser — APPROVED** [review]

**Comment by dschaper** [comment]
Assigned myself for documentation.

**Comment by yubiuser** [comment]
Pinging @dschaper for documentation.

**Comment by yubiuser** [comment]
Documentation: https://github.com/pi-hole/docs/pull/725

**Pull request merged**